okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -894,50 +894,41 @@ public Builder newBuilder() {
   }
 
   /**
-   * Returns a new {@code HttpUrl} representing {@code url} if it is a well-formed HTTP or HTTPS
-   * URL, or null if it isn't.
+   * Returns a new {@code HttpUrl} parsed from the string value of the {@code object} if it is a
+   * well-formed HTTP or HTTPS URL, or null if it isn't.
+   *
+   * @param object for check URL format
+   * @return instance of {@link HttpUrl} or {@code null} in the case of a parse error
    */
-  public static @Nullable HttpUrl parse(String url) {
+  public static @Nullable HttpUrl parse(Object object) {
     Builder builder = new Builder();
-    Builder.ParseResult result = builder.parse(null, url);
+    Builder.ParseResult result = builder.parse(null, String.valueOf(object));
     return result == Builder.ParseResult.SUCCESS ? builder.build() : null;
   }
 
   /**
-   * Returns an {@link HttpUrl} for {@code url} if its protocol is {@code http} or {@code https}, or
-   * null if it has any other protocol.
-   */
-  public static @Nullable HttpUrl get(URL url) {
-    return parse(url.toString());
-  }
-
-  /**
-   * Returns a new {@code HttpUrl} representing {@code url} if it is a well-formed HTTP or HTTPS
-   * URL, or throws an exception if it isn't.
+   * Returns a new {@code HttpUrl} parsed from the string value of the {@code object} if it is a
+   * well-formed HTTP or HTTPS URL, or throws an exception if it isn't.
    *
-   * @throws MalformedURLException if there was a non-host related URL issue
-   * @throws UnknownHostException if the host was invalid
+   * @param object for check URL format
+   * @return instance of {@link HttpUrl}
+   *
+   * @throws IllegalArgumentException if invalid URL specified
    */
-  static HttpUrl getChecked(String url) throws MalformedURLException, UnknownHostException {
+  public static HttpUrl get(Object object) throws IllegalArgumentException {
```

Comment:
I don't know where the comment went but I agree using `Object` here is super weird. We shouldn't be subverting the type system. If we want to support other types we should be supplying overloads.